### PR TITLE
Allow the ability to compute diagnostic score in benchmark

### LINKED ISSUE #311: Allow the ability to compute diagnostic score in a benchmarking run
### Problem Description
The [SDGym benchmark script](https://docs.sdv.dev/sdgym/benchmarking/running-a-benchmark#evaluation) currently allows users to run a quality report for each dataset/synthesizer pair. But it does not allow them to run the diagnostic report.

The diagnostic report (as implemented [in SDMetrics](https://docs.sdv.dev/sdmetrics/reports/diagnostic-report/single-table-api)) checks for overall data validity between the real and synthetic data. For the health of a synthesizer, it is important to verify that the synthetic data it produces always has a score of 1.0.

### Expected behavior
Add a parameter called `compute_diagnostic_score` to the benchmarking script.

- (default) `True`: Run a diagnostic report for each synthesizer/dataset pair being evaluated and report the score. The results should appear in as a new column called `Diagnostic Score`. This column should appear _before_ the `Quality Score` column.
- `False`: Do not run a diagnostic report

### Additional context
When run, the time it takes to compute the diagnostic report should be included in the overall `Evaluate Time`.

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #311 
CU-86b0v4qum

Add diagnostic report before quality report score
Screenshot 2024-06-27 at 7 16 32 PM

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b0v4qum SDGym - Allow the ability to compute diagnostic score in a benchmarking run #311](https://app.clickup.com/t/86b0v4qum)

File: sdgym/benchmark.py [hunk 1]
```diff
@@ -325,15 +353,27 @@ def _score_with_timeout(
...
 
-def _format_output(output, name, dataset_name, compute_quality_score, cache_dir):
-    evaluate_time = None
-    if 'scores' in output or 'quality_score_time' in output:
-        evaluate_time = output.get('quality_score_time', 0)
+def _format_output(
+    output, name, dataset_name, compute_quality_score, compute_diagnostic_score, cache_dir
+):
+    evaluate_time = 0
+    print(output)
```
Comment (gsheni) [inline sdgym/benchmark.py:360]: Remove? 
```suggestion
```

Comment (lajohn4747) [reply sdgym/benchmark.py:360]: I see our lint for this repo doesn't catch print statements lol

Comment (gsheni) [reply sdgym/benchmark.py]: Fix is here:
https://github.com/sdv-dev/SDGym/pull/318

**Review by gsheni — APPROVED** [review]

**Review by amontanez24 — APPROVED** [review]
LGTM!

**Pull request merged**